[New Term for Existing Concept Entry] Pytorch Tensor .size() (#5088)
Codecademy/docs: content/pytorch/concepts/tensors/size/size.md

---
Title: '.size()'
Description: 'Returns the size of the self tensor as a tuple of integers.'
Subjects:
  - 'Machine Learning'
  - 'Data Science'
Tags: 
  - 'AI'
  - 'Data Structures'
  - 'Deep Learning'
  - 'Methods'
CatalogContent: 
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
  - 'paths/data-science-foundations'
---

> The **`size()`** method returns a `torch-size` object containing the size information of a Pytorch tensor. It is a fundamental function used in PyTorch to obtain the tensor's shape and is specifically used for dynamic operations. You can also access specific dimensions using tuples. 

Reviewer: ```suggestion
The **`size()`** method returns a `torch.Size` object containing the size information of a Pytorch tensor. It is a fundamental function used in PyTorch to obtain the tensor's shape and is specifically used for dynamic operations. You can also access specific dimensions using tuples. 
```

## Syntax

```psuedo
t.size(dim=None)
```

> - `dim`: (`int`) This optional parameter specifies the dimension for which to retrieve the size. The default value is `None`. 
>     - If `dim` is not provided, the returned value is a `torch.Size`, a subclass of __tuple__. 
>     - If `dim` is specified, the returned value is an __int__ holding the size of the given dimension. 

Reviewer: ```suggestion
- `dim`: (`int`) This optional parameter specifies the dimension for which to retrieve the size. The default value is `None`. 
    - If `dim` is not provided, the returned value is a `torch.Size`, a subclass of __tuple__. 
    - If `dim` is specified, the returned value is an int holding the size of the given dimension. 
```

## Example

The following example shows how to use the `.size()` method: 

```py
import torch

t = torch.empty(3, 4, 5)

print(t.size())
print(t.size(dim=1))
```

The code above generates the following output:

```shell
torch.Size([3, 4, 5])
4
```

## Codebyte Example

```codebyte/python

> import torch
> 
> t = torch.empty(3, 4, 5)
> 
> print(t.size())
> print(t.size(dim=1))

Reviewer: ```suggestion
import torch

t = torch.empty(3, 4, 5)

print(t.size())
print(t.size(dim=1))
```
Please add another codebyte example.

---
Title: '.size()'
Description: 'Returns the size of the self tensor as a tuple of integers.'
Subjects:
  - 'Machine Learning'
  - 'Data Science'
Tags: 
  - 'AI'
  - 'Data Structures'
  - 'Deep Learning'
  - 'Methods'
CatalogContent: 
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
  - 'paths/data-science-foundations'
---

> The **`size()`** method returns a `torch.Size` object containing the size information of a Pytorch tensor. It is a fundamental function used in PyTorch to obtain the tensor's shape and is specifically used for dynamic operations. You can also access specific dimensions using tuples. 

Reviewer: ```suggestion
The **`size()`** method returns a `torch.Size` object that behaves like a tuple containing the size and information of a PyTorch tensor. This function is a fundamental one used in PyTorch to obtain the tensor's shape. You can also access specific dimensions using index-based access, same as tuples. 
```

---
Title: '.size()'
Description: 'Returns the size of the self tensor as a tuple of integers.'
Subjects:
  - 'Machine Learning'
  - 'Data Science'
Tags: 
  - 'AI'
  - 'Data Structures'
  - 'Deep Learning'
  - 'Methods'
CatalogContent: 
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
  - 'paths/data-science-foundations'
---

The **`size()`** method returns a `torch.Size` object containing the size information of a Pytorch tensor. It is a fundamental function used in PyTorch to obtain the tensor's shape and is specifically used for dynamic operations. You can also access specific dimensions using tuples. 

## Syntax

> ```psuedo
> t.size(dim=None)
> ```

Reviewer: ```suggestion
```pseudo
t.size(dim=None)
```
```
